multicluster/controllers/multicluster/clusterset_controller.go

```diff
@@ -60,3 +122,179 @@ func (r *ClusterSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&multiclusterv1alpha1.ClusterSet{}).
 		Complete(r)
 }
+
+func (r *ClusterSetReconciler) validateLocalClusterClaim(clusterSet *multiclusterv1alpha1.ClusterSet) error {
+	// read the cluster claim in the cluster
+
+	configNamespace := clusterSet.GetNamespace()
+
+	clusterClaimList := &multiclusterv1alpha1.ClusterClaimList{}
+	r.Log.Info("Validating cluster claim in", "namespace", configNamespace)
+	if err := r.List(context.TODO(), clusterClaimList, client.InNamespace(configNamespace)); err != nil {
+		return err
+	}
+	if len(clusterClaimList.Items) == 0 {
+		return fmt.Errorf("cluster claim not configured for the cluster")
+	}
+
+	var clusterSetClaimID string
+	wellKnownClusterSetClaimIDExist := false
+	var clusterClaimID string
+	wellKnownClusterClaimIDExist := false
+	for _, clusterClaim := range clusterClaimList.Items {
+		r.Log.Info("Processing clusterclaim", "name", clusterClaim.Name, "value", clusterClaim.Value)
+		if strings.Compare(clusterClaim.Name, multiclusterv1alpha1.WellKnownClusterClaimClusterSet) == 0 {
+			wellKnownClusterSetClaimIDExist = true
+			clusterSetClaimID = clusterClaim.Value
+		} else if strings.Compare(clusterClaim.Name, multiclusterv1alpha1.WellKnownClusterClaimID) == 0 {
+			wellKnownClusterClaimIDExist = true
+			clusterClaimID = clusterClaim.Value
+		}
+	}
+
+	if !wellKnownClusterSetClaimIDExist {
+		return fmt.Errorf("clusterset claim ID not configured for the cluster")
+	}
+
+	if !wellKnownClusterClaimIDExist {
+		return fmt.Errorf("cluster claim ID not configured for the cluster")
+	}
+
+	configExists := false
+	if r.IsMcsLeader {
+		for _, leader := range clusterSet.Spec.Leaders {
+			if strings.Compare(clusterClaimID, leader.ClusterID) == 0 {
+				configExists = true
+			}
+		}
+		if !configExists {
+			return fmt.Errorf("cluster not defined as leader in cluster set")
+		}
+		// also validate the namespace is the same
+		if strings.Compare(clusterSet.Spec.Namespace, clusterSet.GetNamespace()) != 0 {
+			return fmt.Errorf("cluster set namespace " + clusterSet.Spec.Namespace + " is different from " +
+				clusterSet.GetNamespace())
+		}
+	} else {
+		for _, member := range clusterSet.Spec.Members {
+			if strings.Compare(clusterClaimID, member.ClusterID) == 0 {
+				configExists = true
+			}
+		}
+		if !configExists {
+			return fmt.Errorf("cluster not defined as member in cluster set")
+		}
+	}
+
+	r.clusterID = common.ClusterID(clusterClaimID)
+	r.clusterSetID = common.ClusterSetID(clusterSetClaimID)
+	r.clusterSetConfig = clusterSet.DeepCopy()
```

Comment:
I thought there will be a replica of ClusterSet in member cluster as CR, does this mean it's only in-memory object? how a  user can tell a member cluster which Clusterset it belows to?

Reply:
could you comment on this one?